stock_orderpoint_generator/models/orderpoint_template.py

```diff
@@ -98,12 +96,15 @@ def _template_fields_to_discard(self):
             "auto_max_qty",
         ]
 
-    def _disable_old_instances(self, products):
+    def _remove_old_instances(self, products):
         """Clean old instance by setting those inactives"""
-        orderpoints = self.env["stock.warehouse.orderpoint"].search(
-            [("product_id", "in", products.ids)]
-        )
-        orderpoints.write({"active": False})
+        self.env["stock.warehouse.orderpoint"].search(
+            [
+                ("product_id", "in", products.ids),
+                ("location_id", "=", self.location_id.id),
+                ("company_id", "=", self.company_id.id),
+            ]
+        ).unlink()
```

Comment:
This is a change in the module behavior. @yvaucher

Reply:
Yes, there's a unique key constraint for product, location_id, company_id that won't let us simply archive them. So we can only delete the old ones.